cms/djangoapps/contentstore/tests/test_course_settings.py

```diff
@@ -1106,3 +1109,66 @@ def test_add(self):
         self.assertEqual(obj, grader)
         current_graders = CourseGradingModel.fetch(self.course.id).graders
         self.assertEqual(len(self.starting_graders) + 1, len(current_graders))
+
+
+class CourseMarketingSiteTest(CourseTestCase):
+    """Base class to test course settings details view in Studio when using marketing site flag.
+    """
+    def setUp(self):
+        super(CourseMarketingSiteTest, self).setUp()
+        self.course = CourseFactory.create(org='edX', number='dummy', display_name='Marketing Site Course')
+        self.course_details_url = reverse_course_url('settings_handler', unicode(self.course.id))
+
+    @mock.patch.dict("django.conf.settings.FEATURES", {'ENABLE_MKTG_SITE': False})
+    def test_course_details_with_disabled_setting_global_staff(self):
+        """Test that user enrollment end date is not readonly in response
+        if the feature flag 'ENABLE_MKTG_SITE' is not enabled for global staff..
+        """
+        user = UserFactory()
+        GlobalStaff().add_users(user)
+        self.client.login(username=user.username, password='test')
+
+        response = self.client.get_html(self.course_details_url)
+        self.assertEqual(response.status_code, 200)
+        self.assertNotContains(response, "Contact your edX Partner Manager to update these settings.")
+
+    @mock.patch.dict("django.conf.settings.FEATURES", {'ENABLE_MKTG_SITE': False})
+    def test_course_details_with_disabled_setting_non_global_staff(self):
+        """Test that user enrollment end date is not readonly in response
+        if the feature flag 'ENABLE_MKTG_SITE' is not enabled for non-global staff.
+        """
+        user = UserFactory()
+        for role in [CourseInstructorRole, CourseStaffRole]:
+            role(self.course.id).add_users(user)
+        self.client.login(username=user.username, password='test')
+
+        response = self.client.get_html(self.course_details_url)
+        self.assertEqual(response.status_code, 200)
+        self.assertNotContains(response, "Contact your edX Partner Manager to update these settings.")
+
+    @mock.patch.dict("django.conf.settings.FEATURES", {'ENABLE_MKTG_SITE': True})
+    def test_course_details_with_enabled_setting_global_staff(self):
+        """Test that user enrollment end date is not readonly in response
+        if the feature flag 'ENABLE_MKTG_SITE' is enabled for global staff.
+        """
+        user = UserFactory()
```

Comment:
I would create a helper method for the setup (I really don't like duplicated code).

Something like
def get_course_details_response(global_staff):
    user = UserFactory(is_staff=global_staff)
    self.client.login(username=user.username, password='test')
    response = self.client.get_html(self.course_details_url)
    self.assertEqual(response.status_code, 200)
    return response